odoo/addons/test_inherits/tests/test_inherits.py

```diff
@@ -37,3 +38,37 @@ def test_ir_model_data_inherits(self):
         field = IrModelData.search([('name', '=', 'field_test_box__name')])
         self.assertEqual(len(field), 1)
         self.assertEqual(field.module, 'test_inherits')
+
+    def test_constraint_inherits(self):
+        """Validate constraints on inherits when the parent is not updated"""
+        Model = self.env['test.another_box']
+
+        with self.assertRaises(ValidationError):
+            another_box = Model.create({'val1': 1,'val2': 2})
+
+        another_box = Model.create({'val1': 1,'val2': 1})
+        with self.assertRaises(ValidationError):
+            another_box.write({'val2': 2})
+        another_box.write({'val1': 2, 'val2': 2})
+
+    def test_constraint_inherits_parent_change(self):
+        """Validate constraints on inherits when parent is updated too"""
+        UnitModel = self.env['test.another_unit']
+        BoxModel = self.env['test.another_box']
+
+        unit1 = UnitModel.create({'val1': 1})
+        box = BoxModel.create({'another_unit_id': unit1.id, 'val2': 1})
+
+        unit2 = UnitModel.create({'val1': 2})
+        box.write({'another_unit_id': unit2.id, 'val2': 2})
+
+        unit3 = UnitModel.create({'val1': 3})
+        box.write({'another_unit_id': unit3.id, 'val1': 4, 'val2': 4})
+
+        unit4 = UnitModel.create({'val1': 5})
+        with self.assertRaises(ValidationError):
+            box.write({'another_unit_id': unit2.id, 'val2': 6})
+
+        unit5 = UnitModel.create({'val1': 7})
+        with self.assertRaises(ValidationError):
+            box.write({'another_unit_id': unit2.id, 'val1': 7, 'val2': 8})
```

Comment:
Well, this one cannot not fail, in fact (`val1` always equals 7.)
It would be more interesting to do:
```suggestion
            box.write({'another_unit_id': unit5.id, 'val1': 8, 'val2': 7})
```